theano/gpuarray/ROIPoolGPUFwd.c

```diff
@@ -97,26 +97,38 @@ int APPLY_SPECIFIC(ROIPoolGPUFwd)(PyGpuArrayObject *data,
         return 1;
     }
 
-    if (*out != NULL || *argmaxes != NULL || (!vector_same_shape(data, *out)) || (!vector_same_shape(data, *argmaxes))){
+  if (*out == NULL || !vector_same_shape(data, *out)){
     Py_XDECREF(*out);
-    Py_XDECREF(*argmaxes);
```

Comment:
You should handle both outputs individually. Maybe only one or the other is preallocated, maybe both, maybe none.